In [13]:
%matplotlib inline
from IPython.display import display, clear_output
import pandas as pd
import numpy as np
import time
import gc
import warnings
warnings.filterwarnings('ignore')
opath='~/clinvar_omim_by_GEN/'

In [14]:
clinvar_omim_AR_byGene=pd.read_csv(opath+'AR-clinvarOMIM.tsv',sep='\t')
clinvar_omim_AD_byGene=pd.read_csv(opath+'AD-clinvarOMIM.tsv',sep='\t')
clinvar_omim_BothModels_byGene=pd.read_csv(opath+'ADAR-clinvarOMIM.tsv',sep='\t')

In [32]:
clinvar_omim_AR_byGene.head(30)

Name   Gene(s)  \
0    NM_002645.3(PIK3C2A):c.(?_1-1)_4007+1_4008-1)del   PIK3C2A   
1                      NG_050967.1:g.291668_510915del    EXOC6B   
2                            RFC1, AAGGG(n) EXPANSION      RFC1   
3                                 NM_172862.3:c.15del     FREM2   
4            NM_005236.2(ERCC4):c.388+1164_792+795del     ERCC4   
5                                     RTTN, ARG985GLY      RTTN   
6                  NM_015909.3:c.(?_5028)_(5724_?)del      NBAS   
7                          NM_006642.3:c.1069_1356del   SDCCAG8   
8                                 NM_023343.1:c.55G>A  RNU4ATAC   
9                           NM_000135.2:c.894_2641del     FANCA   
10                          NM_000135.2:c.523_2601del     FANCA   
11  NM_019042.4(PUS7):c.(1757+1_1758-1)_(1848+1_18...      PUS7   
12         NM_001003722.1(GLE1):c.100-7_100-3delTCTCT      GLE1   
13  POLR3A, IVS14, G-A, +22 (rs191875469) AND IVS2...    POLR3A   
14                                   MYORG, ILE655THR     MYORG   
15                                       PRG4, EX1DEL      PRG4   
16                                   HYOU1, ARG419PRO     HYOU1   
17                                   HYOU1, TYR231HIS     HYOU1   
18                    MYO9A, ASP1698GLY (rs150726107)     MYO9A   
19                    MYO9A, ARG2283HIS (rs142345927)     MYO9A   
20                    MYO9A, ARG1517HIS (rs149046541)     MYO9A   
21                       NG_021364.1:g.81245_89166del     WDR66   
22                                RIPK1, 2,064-BP DEL     RIPK1   
23                                   RIPK1, 21-BP DEL     RIPK1   
24                                    RIPK1, 4-BP DEL     RIPK1   
25                                        c.2475-2A-G     KDM5B   
26         NM_000046.4:c.(312+1_313-1_690+1_691-1)del      ARSB   
27                                 PREPL, EX10-15 DEL     PREPL   
28                        GGT1, 16.9-KB DEL/13-BP INS      GGT1   
29                                   TBX19, LYS146ARG     TBX19   

                                         Condition(s)  \
0                         OCULOSKELETODENTAL SYNDROME   
1   SPONDYLOEPIMETAPHYSEAL DYSPLASIA WITH JOINT LA...   
2   Cerebellar ataxia, neuropathy, and vestibular ...   
3                                   FRASER SYNDROME 2   
4                              XFE progeroid syndrome   
5   Microcephaly, short stature, and polymicrogyri...   
6   Short stature, optic nerve atrophy, and Pelger...   
7                            Bardet-Biedl syndrome 16   
8         Osteodysplastic primordial dwarfism, type 1   
9             Fanconi anemia, complementation group A   
10            Fanconi anemia, complementation group A   
11  INTELLECTUAL DEVELOPMENTAL DISORDER WITH ABNOR...   
12  Lethal arthrogryposis with anterior horn cell ...   
13   Neonatal pseudo-hydrocephalic progeroid syndrome   
14  BASAL GANGLIA CALCIFICATION, IDIOPATHIC, 7, AU...   
15  Camptodactyly arthropathy coxa vara pericardit...   
16   Granulocytopenia with immunoglobulin abnormality   
17   Granulocytopenia with immunoglobulin abnormality   
18   MYASTHENIC SYNDROME, CONGENITAL, 24, PRESYNAPTIC   
19   MYASTHENIC SYNDROME, CONGENITAL, 24, PRESYNAPTIC   
20   MYASTHENIC SYNDROME, CONGENITAL, 24, PRESYNAPTIC   
21  SPERMATOGENIC FAILURE 18|SPERMATOGENIC FAILURE 33   
22                                IMMUNODEFICIENCY 57   
23                                IMMUNODEFICIENCY 57   
24                                IMMUNODEFICIENCY 57   
25         MENTAL RETARDATION, AUTOSOMAL RECESSIVE 65   
26                      Mucopolysaccharidosis type VI   
27                Myasthenic syndrome, congenital, 22   
28               gamma-Glutamyltransferase deficiency   
29             Adrenocorticotropic hormone deficiency   

             Clinical significance (Last reviewed)  \
0          Pathogenic(Last reviewed: May 21, 2019)   
1          Pathogenic(Last reviewed: Apr 17, 2019)   
2          Pathogenic(Last reviewed: Apr 

In [16]:
clinvar_recessive=pd.read_csv("/home/felipe/clinvar_result_recessive.txt",sep='\t')
clinvar_dominant=pd.read_csv("/home/felipe/clinvar_result_autosomal_dominant.txt",sep='\t')


In [18]:
clinvar_recessive.head()

Name Gene(s)  \
0               MYORG, ILE655THR   MYORG   
1  NM_020702.5:c.782_783delinsTT   MYORG   
2           NM_003754.2:c.694T>G   EIF3F   
3         NM_032819.4:c.1062delG  ZNF341   
4                   PRG4, EX1DEL    PRG4   

                                        Condition(s)  \
0  BASAL GANGLIA CALCIFICATION, IDIOPATHIC, 7, AU...   
1  BASAL GANGLIA CALCIFICATION, IDIOPATHIC, 7, AU...   
2  INTELLECTUAL DEVELOPMENTAL DISORDER, AUTOSOMAL...   
3  HYPER-IgE RECURRENT INFECTION SYNDROME 3, AUTO...   
4  Camptodactyly arthropathy coxa vara pericardit...   

     Clinical significance (Last reviewed)                   Review status  \
0   Pathogenic(Last reviewed: Feb 8, 2019)  no assertion criteria provided   
1   Pathogenic(Last reviewed: Feb 8, 2019)  no assertion criteria provided   
2  Pathogenic(Last reviewed: Jan 24, 2019)  no assertion criteria provided   
3   Pathogenic(Last reviewed: Feb 4, 2019)  no assertion criteria provided   
4  Pathogenic(Last reviewed: Jan 11, 2019)  no assertion criteria provided   

  GRCh37Chromosome GRCh37Location GRCh38Chromosome GRCh38Location  \
0              NaN            NaN              NaN            NaN   
1              NaN            NaN              NaN            NaN   
2              NaN            NaN              NaN            NaN   
3              NaN            NaN              NaN            NaN   
4              NaN            NaN              NaN            NaN   

   VariationID AlleleID(s)  Unnamed: 11  
0       617697      609100          NaN  
1       617693      609096          NaN  
2       617475      608837          NaN  
3       599412      590749          NaN  
4       599362      590697          NaN

In [25]:
print(len(clinvar_omim_AR_byGene))
for df in [clinvar_recessive,clinvar_dominant]:
    print(sum(clinvar_omim_AR_byGene["VariationID"].isin(df.VariationID)))
    print(len(df))

69809
16889
27227
1900
58713


De un total de 69809 imputadas hay 1900 que en clinvar dice que son dominantes...chequeemos mejor

In [29]:
possible_fails=clinvar_omim_AR_byGene[clinvar_omim_AR_byGene["VariationID"].isin(clinvar_dominant.VariationID)]

In [34]:
possible_fails.clinsig.isin(['Pathogenic','Likely pathogenic']).sum()

148

#### solo 148 de estas variantes parecen patogenicas

In [36]:
possible_fails[possible_fails.clinsig.isin(['Pathogenic','Likely pathogenic'])]

Name  Gene(s)  \
69                                      15q15.3 deletion     STRC   
70                NC_000009.11:g.71705804_71974823invdup     TJP2   
597                               AMHR2, 1-BP DEL, 1692A    AMHR2   
833                                      VPS13A, 8035G-A   VPS13A   
1946          NM_000302.3(PLOD1):c.1533C>G (p.Tyr511Ter)    PLOD1   
2775          NM_156039.3(CSF3R):c.1919C>A (p.Thr640Asn)    CSF3R   
3244   NM_001128425.1(MUTYH):c.1437_1439delGGA (p.Glu...    MUTYH   
3245       NM_001128425.1(MUTYH):c.1438G>T (p.Glu480Ter)    MUTYH   
3269    NM_001128425.1(MUTYH):c.1356delA (p.Tyr453Ilefs)    MUTYH   
3333       NM_001128425.1(MUTYH):c.1214C>T (p.Pro405Leu)    MUTYH   
3343       NM_001128425.1(MUTYH):c.1171C>T (p.Gln391Ter)    MUTYH   
3390       NM_001128425.1(MUTYH):c.1012C>T (p.Gln338Ter)    MUTYH   
3421        NM_001128425.1(MUTYH):c.940C>T (p.Gln314Ter)    MUTYH   
3469        NM_001128425.1(MUTYH):c.799C>T (p.Gln267Ter)    MUTYH   
3486        NM_001128425.1(MUTYH):c.739C>T (p.Arg247Ter)    MUTYH   
3515          NM_001128425.1(MUTYH):c.690G>A (p.Gln230=)    MUTYH   
3535        NM_001128425.1(MUTYH):c.628C>T (p.Gln210Ter)    MUTYH   
3548        NM_001128425.1(MUTYH):c.586G>T (p.Glu196Ter)    MUTYH   
3549                    NM_001128425.1(MUTYH):c.577-2A>G    MUTYH   
3572        NM_001128425.1(MUTYH):c.545G>A (p.Arg182His)    MUTYH   
3578        NM_001128425.1(MUTYH):c.536A>G (p.Tyr179Cys)    MUTYH   
3592                    NM_001128425.1(MUTYH):c.505-2A>C    MUTYH   
3601                    NM_001128425.1(MUTYH):c.504+2T>C    MUTYH   
3615        NM_001128425.1(MUTYH):c.463A>T (p.Lys155Ter)    MUTYH   
3678        NM_001128425.1(MUTYH):c.312C>A (p.Tyr104Ter)    MUTYH   
3682        NM_001128425.1(MUTYH):c.308G>A (p.Trp103Ter)    MUTYH   
3686         NM_001128425.1(MUTYH):c.289C>T (p.Arg97Ter)    MUTYH   
3733       NM_001128425.1(MUTYH):c.91delG (p.Ala31Profs)    MUTYH   
3747          NM_001128425.1(MUTYH):c.55C>T (p.Arg19Ter)    MUTYH   
5540            NM_001232.3(CASQ2):c.164A>G (p.Tyr55Cys)    CASQ2   
...                                                  ...      ...   
60520  NM_058216.2(RAD51C):c.186_187delAA (p.Gln62Hisfs)   RAD51C   
60533        NM_058216.3(RAD51C):c.224dup (p.Tyr75Terfs)   RAD51C   
60543          NM_058216.2(RAD51C):c.250A>T (p.Lys84Ter)   RAD51C   
60586      NM_058216.2(RAD51C):c.394dupA (p.Thr132Asnfs)   RAD51C   
60590         NM_058216.2(RAD51C):c.397C>T (p.Gln133Ter)   RAD51C   
60594         NM_058216.2(RAD51C):c.404G>C (p.Cys135Ser)   RAD51C   
60600                     NM_058216.2(RAD51C):c.405-1G>C   RAD51C   
60604         NM_058216.2(RAD51C):c.414G>C (p.Leu138Phe)   RAD51C   
60637         NM_058216.2(RAD51C):c.502A>T (p.Arg168Ter)   RAD51C   
60660         NM_058216.2(RAD51C):c.577C>T (p.Arg193Ter)   RAD51C   
60670         NM_058216.2(RAD51C):c.630T>G (p.Tyr210Ter)   RAD51C   
60687         NM_058216.2(RAD51C):c.701C>G (p.Ser234Ter)   RAD51C   
60689                     NM_058216.2(RAD51C):c.705+1G>A   RAD51C   
60702         NM_058216.2(RAD51C):c.709C>T (p.Arg237Ter)   RAD51C   
60726         NM_058216.2(RAD51C):c.773G>A (p.Arg258His)   RAD51C   
60743                     NM_058216.2(RAD51C):c.837+1G>T   RAD51C   
60744                     NM_058216.2(RAD51C):c.837+1G>A   RAD51C   
60749                     NM_058216.2(RAD51C):c.838-2A>G   RAD51C   
60768  NM_058216.2(RAD51C):c.890_899delTTGTTCCTGC (p....   RAD51C   
60774                     NM_058216.2(RAD51C):c.904+5G>T   RAD51C   
60779                     NM_058216.2(RAD51C):c.905-2A>G   RAD51C   
60780                    NM_058216.2(RAD51C):c.905-2delA   RAD51C   
60792         NM_058216.2(RAD51C):c.955C>T (p.Arg319Ter)   RAD51C   
61095                     NM_018672.4(ABCA5):c.4320+1G>C    ABCA5   
65231        NM_016941.3(DLL3):c.1440del (p.Pro481Argfs)     DLL3   
66274         NM_153638.3(PANK2):c.1561G>A (p.Gly521Arg)    PANK2   
67048        NM_030777.3(SLC2A10):c.685C>T (p.Arg2

hay varias de estas con criterios de mucho peso habria que chequear bien que esta pasando con el cruce y si justo los genes en que caen estas no habria que transformarlos a herencia multiple 


In [26]:
print("total dominant cruce")
print(len(clinvar_omim_AD_byGene))
for df in [clinvar_recessive,clinvar_dominant]:
    print(sum(clinvar_omim_AD_byGene["VariationID"].isin(df.VariationID)))
    print(len(df))

76068
199
27227
15222
58713


In [38]:
possible_fails=clinvar_omim_AD_byGene[clinvar_omim_AD_byGene["VariationID"].isin(clinvar_recessive.VariationID)]

In [39]:
possible_fails.clinsig.isin(['Pathogenic','Likely pathogenic']).sum()

107

solo 107 de estas variantes parecen patogenicas

In [40]:
possible_fails[possible_fails.clinsig.isin(['Pathogenic','Likely pathogenic'])]

Name  Gene(s)  \
741                                CRX, 2-BP DEL, GLU168      CRX   
751                                         OPTN, EX5DEL     OPTN   
830                              FOXL2, 15-BP DUP, NT684    FOXL2   
1389        NM_005026.4(PIK3CD):c.3061G>A (p.Glu1021Lys)   PIK3CD   
1976             NM_003000.2(SDHB):c.268C>T (p.Arg90Ter)     SDHB   
2008             NM_003000.2(SDHB):c.143A>T (p.Asp48Val)     SDHB   
2158       NM_001020658.1(PUM1):c.3439C>T (p.Arg1147Trp)     PUM1   
2159       NM_001020658.1(PUM1):c.3415C>T (p.Arg1139Trp)     PUM1   
2160       NM_001020658.1(PUM1):c.3103A>T (p.Thr1035Ser)     PUM1   
3175            NM_002016.1(FLG):c.2143C>T (p.Gln715Ter)      FLG   
3439            NM_003001.3(SDHC):c.397C>T (p.Arg133Ter)     SDHC   
3575             NM_000261.1(MYOC):c.136C>T (p.Arg46Ter)     MYOC   
3983       NM_000069.2(CACNA1S):c.3716G>A (p.Arg1239His)  CACNA1S   
6642           NM_004522.2(KIF5C):c.709G>A (p.Glu237Lys)    KIF5C   
9411             NM_212482.3(FN1):c.778T>G (p.Cys260Gly)      FN1   
9413             NM_212482.3(FN1):c.718T>G (p.Tyr240Asp)      FN1   
9415             NM_212482.3(FN1):c.675C>G (p.Cys225Trp)      FN1   
9419             NM_212482.3(FN1):c.367T>C (p.Cys123Arg)      FN1   
9420              NM_212482.3(FN1):c.260G>T (p.Cys87Phe)      FN1   
10469  NM_001904.3(CTNNB1):c.1272_1275delTTCT (p.Ser4...   CTNNB1   
12424         NM_016306.5(DNAJB11):c.161C>G (p.Pro54Arg)  DNAJB11   
12426         NM_016306.5(DNAJB11):c.230T>C (p.Leu77Pro)  DNAJB11   
12428        NM_016306.5(DNAJB11):c.616C>T (p.Arg206Ter)  DNAJB11   
13617        NM_000297.3(PKD2):c.670delC (p.Leu224Trpfs)     PKD2   
13924         NM_018359.4(UFSP2):c.1277A>C (p.Asp426Ala)    UFSP2   
18765         NM_000358.2(TGFBI):c.1501C>A (p.Pro501Thr)    TGFBI   
19124        NM_002609.3(PDGFRB):c.1681C>T (p.Arg561Cys)   PDGFRB   
19167  NM_001135243.1(TCOF1):c.1637_1640delAGAG (p.Gl...    TCOF1   
19219       NM_001037333.3(CYFIP2):c.260G>C (p.Arg87Pro)   CYFIP2   
19220       NM_001037333.3(CYFIP2):c.260G>T (p.Arg87Leu)   CYFIP2   
...                                                  ...      ...   
53650           NM_004960.3(FUS):c.1551C>G (p.His517Gln)      FUS   
55479        NC_000016.10:g.(?_23621352)_(23641167_?)del    PALB2   
55519           NC_000016.9:g.(?_2152062)_(2152634_?)del     PKD1   
55520            NM_001009944.2(PKD1):c.8017-?_8161+?dup     PKD1   
55521           NC_000016.9:g.(?_2149645)_(2158700_?)del     PKD1   
55522                NM_024675.3(PALB2):c.-200_3113+?del    PALB2   
60304         NM_000088.3(COL1A1):c.994G>A (p.Gly332Arg)   COL1A1   
61031  NM_032043.2(BRIP1):c.2108delAinsTCC (p.Lys703I...    BRIP1   
61230         NM_032043.2(BRIP1):c.1383T>G (p.Tyr461Ter)    BRIP1   
61680          NM_001284333.1(TLK2):c.37C>T (p.Gln13Ter)     TLK2   
61681         NM_001284333.1(TLK2):c.181C>T (p.Arg61Ter)     TLK2   
61682         NM_001284333.1(TLK2):c.202G>T (p.Glu68Ter)     TLK2   
61684           NM_006852.3(TLK2):c.784C>T (p.Arg262Ter)     TLK2   
61686           NM_006852.3(TLK2):c.907C>T (p.Arg303Ter)     TLK2   
61687           NM_006852.3(TLK2):c.989C>A (p.Ser330Ter)     TLK2   
61691       NM_001284333.1(TLK2):c.1478A>G (p.His493Arg)     TLK2   
61693       NM_001284333.1(TLK2):c.1553A>G (p.His518Arg)     TLK2   
61694       NM_001284333.1(TLK2):c.1702C>T (p.Arg568Trp)     TLK2   
61695       NM_001284333.1(TLK2):c.1717C>T (p.Gln573Ter)     TLK2   
61697       NM_001284333.1(TLK2):c.1885G>A (p.Asp629Asn)     TLK2   
61698       NM_001284333.1(TLK2):c.2039C>G (p.Pro680Arg)     TLK2   
63625        NM_015559.3(SETBP1):c.2612T>C (p.Ile871Thr)   SETBP1   
67855            NM_000527.4(LDLR):c.1216C>A (p.Arg406=)     LDLR   
69708         NM_014727.2(KMT2B):c.1690C>T (p.Arg564Ter)    KMT2B   
69709         NM_014727.2(KMT2B):c.2428C>T (p.Gln810Ter)    KMT2B   
69710        NM_014727.2(KMT2B):c.4545C>A (p.Tyr1515Ter)    KMT2B   
69849        NM_152296.5(ATP1A3):c.2401G>A (p.Asp8